src/Microsoft.AspNet.PageExecutionInstrumentation.Interfaces/IBufferedTextWriter.cs

```diff
@@ -19,14 +20,16 @@ public interface IBufferedTextWriter
         /// <summary>
         /// Copies the buffered content to the <paramref name="writer"/>.
         /// </summary>
-        /// <param name="writer">The writer to copy the contents to.</param>
-        void CopyTo(TextWriter writer);
+        /// <param name="writer">The <see cref="TextWriter"/> to copy the contents to.</param>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/> which encodes the content to be written.</param>
+        void CopyTo(TextWriter writer, IHtmlEncoder encoder);
 
         /// <summary>
         /// Asynchronously copies the buffered content to the <paramref name="writer"/>.
         /// </summary>
-        /// <param name="writer">The writer to copy the contents to.</param>
+        /// <param name="writer">The <see cref="TextWriter"/> to copy the contents to.</param>
+        /// <param name="encoder">The <see cref="IHtmlEncoder"/> which encodes the content to be written.</param>
         /// <returns>A <see cref="Task"/> representing the copy operation.</returns>
-        Task CopyToAsync(TextWriter writer);
+        Task CopyToAsync(TextWriter writer, IHtmlEncoder encoder);
```

Comment:
I have an MVC6 intranet application that uses roles. The admin users can
manage all the user roles. However any changes made to a users' roles are
only effective when the user logs on (this makes sense as the roles are
stored as claims in a cookie). The requirement is for role changes to take
effect immediately, like they did in the older asp.net membership and role
managers. How can one achieve this with the new identity models? The only
way that comes to mind is to set some flag that is checked on each request
and if set then update the cookie to reflect the new roles. Do you have any
other suggestions or examples of how to ensure the role changes take effect
immediatley?

On Thu, Jul 9, 2015 at 1:14 PM, Pranav K <notifications@github.com> wrote:

> In
> src/Microsoft.AspNet.PageExecutionInstrumentation.Interfaces/IBufferedTextWriter.cs
> <https://github.com/aspnet/Mvc/pull/2797#discussion_r34272555>:
>
> >          /// <returns>A <see cref="Task"/> representing the copy operation.</returns>
> > -        Task CopyToAsync(TextWriter writer);
> > +        Task CopyToAsync(TextWriter writer, IHtmlEncoder encoder);
>
> We'll need to let the BrowserLink folks know if any API changes are made
> in this project.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aspnet/Mvc/pull/2797/files#r34272555>.
>